### ISSUE TITLE
performance of training a new transformer

### ISSUE THREAD

**Issue by zhao1402072392** [pytorch/fairseq#1923]
i follow the commend of training a new model(here is transformer)on iwslt2014 de-en, but I just got BLEU score 7.94 at 80th epoch.

my commend are as follows:

cd examples/translation/
bash prepare-iwslt14.sh
cd ../..
TEXT=examples/translation/iwslt14.tokenized.de-en
python3 preprocess.py --source-lang de --target-lang en --trainpref $TEXT/train --validpref $TEXT/valid --testpref $TEXT/test --destdir data-bin/iwslt14.tokenized.de-en

mkdir -p checkpoints/transformerlr0001
nohup python3 train.py data-bin/iwslt14.tokenized.de-en --lr 0.003 --clip-norm 0.1 --dropout 0.3 --batch-size 50 --arch transformer --no-progress-bar --save-dir checkpoints/transformerlr0003 > transformerlr0003.out 2>&1 &

python3 generate.py data-bin/iwslt14.tokenized.de-en --path checkpoints/transformerlr0001/checkpoint_best.pt   --beam 5 --batch-size 128 --remove-bpe | tee ./tmp/gen.out

Could you give me some advice, thanks very much ^_^

**Comment by erip** [comment]
It doesn't seem like you're training/applying BPE to your input sequences.

**Comment by zhao1402072392** [comment]
i think i did, beacause that the \fairseq\examples\translation\prepare-iwslt14.sh applying BPE,and I saw my processed dataset ,it's indeed applying BPE.

**Comment by erip** [comment]
Oops - I didn't see that in your commands above! You should make `--max-tokens` as large as possible without blowing out memory when training this model. Replace `--batch-size` with `--max-tokens 4096` in your training step.

**Comment by zhao1402072392** [comment]
ok,i'll try that, thank u very much ^_^

**Comment by huihuifan** [comment]
Agree with @erip's advice. It looks like your command should work with this fix. Other things to look out for are adjusting the learning rate and dropout if your dataset is very different in size than IWSLT. Please open another issue if you are having trouble, thanks!
